solr/contrib/ltr/src/java/org/apache/solr/response/transform/LTRFeatureLoggerTransformerFactory.java

```diff
@@ -186,18 +185,17 @@ public void setContext(ResultContext context) {
 
       featureLogger = reRankModel.getFeatureLogger();
 
-      Weight w;
+      ModelWeight w;
       try {
         w = reRankModel.createWeight(searcher, true, 1f);
       } catch (final IOException e) {
         throw new SolrException(ErrorCode.BAD_REQUEST, e.getMessage(), e);
       }
-      if ((w == null) || !(w instanceof ModelWeight)) {
+      if (w == null) {
         throw new SolrException(ErrorCode.BAD_REQUEST,
             "error logging the features, model weight is null");
       }
-      modelWeight = (ModelWeight) w;
-
+      modelWeight = w;
```

Comment:
Do we need this line anymore, or can you just do modelWeight = createWeight(...) in the try?